### PR TITLE
Add publishing workflow and contribution instructions

### PR DESCRIPTION
**Pull request by pederhan**
Tasks:

- [x] Add changelog
- [x] Add contributing instructions
- [x] Adapt tox environments for building and publishing
- [x] Add workflow for building and publishing (see under)

Actions workflow:
- [x] Build source+wheels
- [x] Build binaries
- [x] Publish source+wheels to PyPI
- [x] Create GitHub release
- [x] Add source+wheels to release
- [x] Add binaries to release

### CODE REVIEW

**Comment by pederhan** [comment]
I deleted the resulting releases (should have waited until you had a chance to see it), but I can confirm that the workflow automatically creates a new GitHub release with binaries for Windows, MacOS (ARM) and Linux, as well as source and wheel distributions. Publishing to PyPI happens through a [Trusted Publisher](https://docs.pypi.org/trusted-publishers/) workflow.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+jobs:
+  build_pypi:
+    name: Build wheels and source distribution
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Only release from master branch
+      # https://stackoverflow.com/a/74690436
```
Comment (terjekv) [inline .github/workflows/publish.yml:20]: If we create a branch, `notmaster`, that'd trigger it? Not that `/master` is appealing.

File: CHANGELOG [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+### Added
+
+- `--version` option to display current version and exit. 
+    Version number is in the form of `major.minor.patch` if installed from a published version.
+    Shows a version number including the commit hash if installed from a git repository in the form of `major.minor.patch.dev123+gabc1234`.
+    See [Default versioning scheme](https://setuptools-scm.readthedocs.io/en/latest/usage/#default-versioning-scheme) in the [setuptools_scm](https://github.com/pypa/setuptools_scm/) documentation for more information.
+- `label set_description` command to set the description of a label.
+- `network list_excluded_ranges` command to list the excluded ranges of a network.
+- Application can now store tokens for multiple servers and will pick the correct one based on the server URL.
+
```
Comment (terjekv) [inline CHANGELOG:21]: Add a line about adding artifact builds to Github and pypi for releases?

File: CHANGELOG [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+### Changed
+
+- The application now uses Pydantic internally to validate request and response data. This should make the code more robust and easier to maintain.
+- `label rename` no longer takes a `-desc` option. Description modification is now done with the `label set_description` command.
+- Application now attempts to send JSON for every request. This should improve the consistency of the API responses.
+
+### Fixed
+
+- Hopefully more than we broke.
```
Comment (terjekv) [inline CHANGELOG:31]: That's a low, low bar. :)

File: .github/workflows/publish.yml (hunk 1 above)
Comment (pederhan) [reply .github/workflows/publish.yml]: Fixed in https://github.com/unioslo/mreg-cli/pull/254/commits/701ce7ac709910c26b12345d29e8bbaf42fed259

See the _Dump job github var_ step https://github.com/unioslo/mreg-cli/actions/runs/9445593618/job/26013521198 for other variables we can use to determine this...

File: CHANGELOG (hunk 3 above)
Comment (pederhan) [reply CHANGELOG:36]: But it's a bar 😎

**Comment by pederhan** [comment]
Leaving the final release up so we can use that as a starting point for further iteration in the future:

https://github.com/unioslo/mreg-cli/releases
https://pypi.org/project/mreg-cli/#description

I definitely should have published this to Test PyPI instead of the regular one, but that little mistake at least helped us verify that the trusted publishing workflow works for regular PyPI as well.

Here's the package on Test PyPI: https://test.pypi.org/project/mreg-cli/

**Pull request merged**